src/snowflake/snowpark/_internal/data_source/utils.py

```diff
@@ -74,6 +82,7 @@ class DRIVER_TYPE(str, Enum):
         "msodbcsql",
         "snowflake-snowpark-python",
     ],
+    DBMS_TYPE.MYSQL_DB: ["pymysql>=1.0.0,<=2.0.0", "snowflake-snowpark-python"],
```

Comment:
major version bump can introduce breaking changes, it should be less than 2
```suggestion
    DBMS_TYPE.MYSQL_DB: ["pymysql>=1.0.0,<2.0.0", "snowflake-snowpark-python"],
```